source/features/rgh-improve-new-issue-form.tsx

```diff
@@ -5,12 +5,42 @@ import * as pageDetect from 'github-url-detection';
 import features from '../feature-manager';
 import openOptions from '../helpers/open-options';
 import clearCacheHandler from '../helpers/clear-cache-handler';
-import {expectTokenScope} from '../github-helpers/api';
+import {expectToken, expectTokenScope} from '../github-helpers/api';
 import {isRefinedGitHubRepo} from '../github-helpers';
+import {assertNodeContent} from '../helpers/dom-utils';
 
-function init(): void {
+function addNotice(adjective: JSX.Element | string): void {
+	select('#issue_body_template_name')!.before(
+		<div className="flash flash-warn m-2">{}
+			Your Personal Access Token is {adjective}. Some Refined GitHub features will not work without it.
+			You can update it <button className="btn-link" type="button" onClick={openOptions as unknown as React.MouseEventHandler}>in the options</button>.
+		</div>,
+	);
+}
+
+async function checkToken(): Promise<void> {
+	try {
+		await expectToken();
+	} catch {
+		addNotice('missing');
+		return;
+	}
+
+	try {
+		await expectTokenScope('repo');
+	} catch {
+		addNotice('invalid or expired');
+	}
+}
+
+async function setVersion(): Promise<void> {
 	const {version} = browser.runtime.getManifest();
-	select('input#issue_form_extension_version')!.value = version;
+	const versionFieldId = 'issue_form_4ceb28848d92a2c2094093043f57e5e4cb152518a3c2b3b950d1273f795dc13c';
+	assertNodeContent(select(`label[for="${versionFieldId}"`)!.firstChild, 'Extension version');
+	select(`input#${versionFieldId}`)!.value = version;
```

Comment:
Dynamic IDs. Hopefully they're actually hashes of user content rather than totally random.

This was breaking each part of the feature, so I refactored it to avoid this issue in the future

Reply:
Fixed in aed513d